file-service-persistence/src/main/java/uk/gov/justice/services/fileservice/repository/FileStore.java

```diff
@@ -62,30 +70,84 @@ public void store(
      * for that id then {@link Optional}.empty() is returned.
      *
      * If no metadata is found for that id then {@code empty()} is returned
-     * If metadata is found but no content, then an {@link TransactionFailedException} is thrown
      *
      * @param fileId the id of the file
-     * @param connection a database connection
-     * @return if found: the {@link StorableFile} for that id wrapped in an {@link Optional}
      * @return if not found: {@link Optional}.empty()
-     * @throws TransactionFailedException if any of the database updates failed and the transaction
-     * should be rolled back
      */
-    public Optional<StorableFile> find(final UUID fileId, final Connection connection) throws TransactionFailedException {
+    @Transactional
+    public Optional<FileReference> find(final UUID fileId) {
+
+        final Connection connection = connectionProvider.getConnection();
+
+        try {
+            final Optional<JsonObject> metadata = metadataJdbcRepository.findByFileId(fileId, connection);
+            final Optional<InputStream> content = contentJdbcRepository.findByFileId(fileId, connection);
+
+            if (!metadata.isPresent()) {
+                return empty();
+            }
+            if (!content.isPresent()) {
```

Comment:
What if context exists but the metadata doesn't? this error handling is slanted only to the metadata.
I think we need to throw an exception here too as it is consistent and the metadata should be mandatory.